joss/paper.md

```diff
@@ -0,0 +1,105 @@
+
+The framework's distinct approach of requiring only single-operation "site selection" with no metadata storage provides significant improvements in space efficiency compared to previous methods. This makes Downstream especially valuable for applications where individual data items may be as small as single bits or bytes—scenarios, where conventional approaches would consume more memory for metadata than for the data itself.
+
+
+# Methods
+
+To simplify the process of stream curation, this approach makes some assumptions: first, the output buffer is of a fixed length; second, stored data remains static without subsequent editing or relocation. Operations on ingested items are therefore limited to discarding without storage, storing, or overwriting previously stored data. This reduces the problem of stream curation to one of site selection—determining the optimal placement of each subsequent data item.
+
+To support diverse use cases, the three Downstream algorithms address different temporal distributions.
+
+The steady algorithm creates evenly distributed snapshots across stream history by preserving data points with the most historical significance while maintaining relatively uniform spacing between retained items. This approach is best suited for applications in which it is important to maintain data from all time periods, such as long-term monitoring systems that need to detect patterns across their entire operational history [@microsoft_monitoring] and trend analysis across extended timeframes [@techtarget_monitoring].
+
+The stretched algorithm prioritizes older data while maintaining recent context, focusing on preserving the origins of the stream. This approach particularly benefits applications where understanding initial conditions is critical for providing context for current behaviors, such as system diagnostics, evolutionary patterns [@moreno2024algorithms], and environmental monitoring systems [@researchgate_monitoring].
+
+The tilted algorithm prioritizes recent information over older data. This makes it well-suited for monitoring and alerting systems where recent trends are most relevant, but historical context still provides valuable perspective such as in real-time monitoring systems [@tabassum2018sampling] where recent data carries more operational significance than older data [@aggarwal2006biased].
+
+# Implementation
+
+Downstream has been implemented across five programming languages in order to support diverse computing environments, including C++, Rust, Python, Zig, and the Cerebras Software Language. Each of the implementations reside in a separate branch of the Downstream GitHub repository and contains site selection implementation for the three main Downstream algorithms. In addition, some branches include hybrid variants. To enhance usability and reliability, the framework incorporates features across all implementations:
```

Comment:
We should probably hint why the support across programming languages is important.